core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java

```diff
@@ -254,24 +249,25 @@ private ContentPreviewResponse contentPreviewResponse(HttpResponseDuplicator res
 
     private static long getRetryAfterMillis(ClientRequestContext ctx) {
         final HttpHeaders headers = firstNonNull(ctx.log().responseHeaders(), HttpHeaders.of());
-        long millisAfter = -1;
         final String value = headers.get(HttpHeaderNames.RETRY_AFTER);
         if (value != null) {
             try {
-                millisAfter = Duration.ofSeconds(Integer.parseInt(value)).toMillis();
-                return millisAfter;
+                return Duration.ofSeconds(Integer.parseInt(value)).toMillis();
             } catch (Exception ignored) {
                 // Not a second value.
             }
-            try {
-                final Long later = headers.getTimeMillis(HttpHeaderNames.RETRY_AFTER);
-                millisAfter = later - System.currentTimeMillis();
-            } catch (Exception ignored) {
-                logger.debug("The retryAfter: {}, from the server is neither an HTTP date nor a second.",
-                             value);
+
+            @SuppressWarnings("UseOfObsoleteDateTimeApi")
+            final Date date = DateFormatter.parseHttpDate(value);
+            if (date != null) {
```

Comment:
I wonder we don't need to catch `IllegalArgumentException`.
IIUC, it can throw `IllegalArgumentException` even though [the documentation](https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/DateFormatter.java#L102) says it returns null if it fails to be parsed.

Reply:
Oh, indeed. I fixed everywhere that calls `parseHttpDate()`.